### ISSUE TITLE
Continuous unstoppable tone produced on LV2, VST3 plug-in versions (qtractor, jalv)

### ISSUE THREAD

**Issue by rncbc** [iurie-sw/geonkick#125]
As soon a note-on is triggered, or the "play" button is hit on the GUI, a continuous, accumulative tone appears on 1st pair of audio outputs.

a. only the plugin versions are affected; the standalone binary does not appear to suffer from this nasty behavior;
b. seems to happen on qtractor (LV2, VST3) and jalv (LV2);
c. ardour (v5 and 6) seems to be immune to this symptom (LV2).

builds tested on openSUSE TW x86_64; LV2 1.18; VST 3.7.0; all from current git master branch HEAD.

**Comment by iurienistor** [comment]
Hi! Thank you, I'll check to see if I can reproduce.

**Comment by rncbc** [comment]
one other thing re. VST3 plugin: geonkick.vst3 seems to miss  `ModuleEntry` and `ModuleExit ` calls; as far as the original VST3 SDK mandates.

**Comment by iurienistor** [comment]
I have seen this behaviour also with jack. The sound actually is repetitive. It can be fixed if I am memseting the buffer before using it every time the audio callback is called. It looks like the host is not zeroing the buffer and I don't know if it should or not. 

For LV2 and VST3 I am not zeroing myself before I am using it. I have tested now and if I am zeroing than the issue goes away.

**Comment by rncbc** [comment]
on my book **output** buffers must be zeroed (and they are zeroed by the host befora handing out to the plugin at least in qtractor case).

but note, that is *only* geonkick (.lv2, .vst3) plugins that are showing this behavior; all others, from years to date, and all other authors and vendors do simply do this--you're the *special one* ;)

**Comment by iurienistor** [comment]
I've just made some logs and the buffer that comes from qtractor is not zeroed and it contains the data. In Ardour is zeroed.

Also, tested the issue in Ardour/LV2, REAPER/VST3 and MusE/LV2 and can't reproduce. In Bitwig/VST3 is reproduced. The issue for qtractor and Bitwig is fixed if I am zeroing the buffer before use.

If Geonkick for some reason generates this looped sound than this can't be fixed just by just zeroing the buffer that comes from the host before using it. After zeroing Geonkick would fill it with the sound.

The idea is that I am adding in the buffer until keyoff. Than I am adding zero to buffer values after of key off. If the buffer values that comes from the host are not zeroed, Geonkick in this way will not touch them, thus,  they will go back to be played by the host... this creates a never ending looped sound even after key off.

If this is not the case and Geonkick is doing something other things wrong than my hypothesis is Geonkick might have a buffer overflow and it writes in another host buffer before the audio callback is ended and than the buffer will be sent back again to the plugin ... but I hardly imagine that this will result in this neverending looped sound.

Probably plugins usually touch all values of the buffer even if there is no key on by setting them to zero... they don't add zero, that's why most of the plugins probably won't experinece this problem.

Anyway, I have fixed the issue, but I will investigate more on this. The issue is it in the master branch and soon I'll release a patch. Thank you for this important issue report. :)
